torch/_inductor/codegen/triton.py

```diff
@@ -2729,6 +2747,68 @@ def select_tiling(cls, node_schedule, numel, reduction_numel=sympy.Integer(1)):
     def flush(self):
         pass
 
+    def benchmark_fused_nodes(self, nodes):
+        _, (numel, rnumel) = max(nodes, key=lambda x: int(x.is_reduction())).group
+        node_schedule = self.generate_node_schedule(nodes, numel, rnumel)
+        tiled_groups = self.select_tiling(node_schedule, numel, rnumel)
+        reduction_hint_val, mutations, index_dtype = self.get_kernel_args(
+            node_schedule, numel, rnumel
+        )
+
+        kernel = TritonKernel(
+            *tiled_groups,
+            reduction_hint=reduction_hint_val,
+            mutations=mutations,
+            index_dtype=index_dtype,
+        )
+
+        # empty last_usage. May cause more aggressive 'evict_last'. Should be fine.
+        for n in nodes:
+            n.last_usage = set()
+
+        self.codegen_node_schedule_with_kernel(node_schedule, kernel)
+        with config.patch("benchmark_kernel", True), V.set_kernel_handler(kernel):
+            src_code = kernel.codegen_kernel()
+        mod = PyCodeCache.load(src_code)
+
+        def cache_file_path():
+            return os.path.splitext(mod.__file__)[0] + ".kernel_perf"
```

Comment:
Do you think it would be possible to rely on the `AlgorithmSelectorCache` here instead (not necessarily in this PR, but maybe in a follow-up), the same as the autotuning currently relies on? Then we could automatically reap the benefits of autotuning improvements: centralized caching, parallelized autotuning (currently WIP), managed caching (e.g., for internal / prod use cases). Also, I believe, that would open the door for easier "autotuning + epilogue fusion" integration.

Reply:
Hmm, `AlgorithmSelectorCache` has a structure designed for MM tuning: e.g. tensor shapes and benchmark results are stored in the cache. While benchmark fusion here should be generally applicable to pw/reduction/template. So I don't find it straightforward to directly reuse `AlgorithmSelectorCache` here, unless you see some good ways 

On the other hand, both AlgorithmSelectorCache and the code here rely on `CodeCache` class. So I think a `managed caching` (I assume this as a cache shared by multiple machines) can still be done in `CodeCache` level.

Also the workload here is not as easy parallelizable as mm tuning. For mm tuning, we know before handle a bunch of triton configs to parallelize. Here we don't. But I can image with some redesign, we can batch and parallelize the benchmarking requests.

Reply:
Thanks for the clarification, makes sense. The reason I've asked about unification with the `AlgorithmSelectorCache` is that I was imagining that, at a later point, we could defer autotuning to be done in an epilogue fusion-aware way: e.g., select the best option from autotune choices with and without epilogue fusion. And then there would be plently of parallelism, too. This would likely require bringing `AlgorithmSelectorCache` into the picture started in this PR, but maybe it's too abstract at this point.

Reply:
Actually, `AlgorithmSelectorCache` relies of `PersistentCache`, while in this PR `PyCodeCache` is used. My understanding is that those are disjoint. So maybe we'll need to consider a managed `PyCodeCache` location, too.

Reply:
Good catch yea. Maybe we need some refactoring to do the decision of cache location in some sort of base class